File: watchweb.js
// WatchWeb.js | NodeJS implementation of WatchWeb
// Author - ScaredOS | 4/16/2020
// Usage: node watchweb.js <url> <filename>
'use strict';
var fs = require('fs');
var https = require('https');
var util = require('util');
var url = process.argv[2];
var filename = process.argv[3];
var data = '';


function get() {
  https.get(url, res => {
    data = util.format("[%s] Response code on (%s) - %s\n", res.statusCode, url, new Date().toISOString().replace(/T/, ' ').replace(/\..+/, ''));
    // console.log(data); // Uncomment this line to log the data to console
    fs.appendFile(filename, data, function (err) {
      if (err) throw err;
    });
  });
}

get();
setInterval(get, 300000);

File: watchweb.dart
// WatchWeb.dart | Dart Implementation of WatchWeb
// Author - ScaredOS | 4/24/2020
// Usage: dart watchweb.dart <url> <filename>
import 'dart:io';
import 'dart:convert';
import 'package:intl/intl.dart';
import 'package:http/http.dart' as http;

main(List<String> args) async {
  int x = 1;
  var url = args[0];
  var fil = args[1];
  var file = new File(fil);
  var date = new DateFormat('yyyy-MM-dd hh:mm:ss');
  var sink = file.openWrite();
  while(x < 2) {
    var res = await http.get(url);
    //print('[${res.statusCode}] Response code on ($url) - ${date.format(new DateTime.now())}'); // Uncomment to print to console
    sink.write('[${res.statusCode}] Response code on ($url) - ${date.format(new DateTime.now())}\n');
    sleep(const Duration(seconds:1));
  }
}
